### PR TITLE
Add condition form class and create action

### PR DESCRIPTION
**Pull request by DavidBiddle**
#### What problem does the pull request solve?

- Adds a new `Pages::ConditionsForm` class for validations and custom functionality needed on the new conditions page
- Add new `create` post endpoint to create the actual condition record
- basic form validations on the new conditions pages
- adds default nil options to the answer_value and goto_page_id select list
- adds a "None of the above" option to answer_value if the page is optional

Trello card: https://trello.com/c/gxgmZmXA/564-add-conditions-to-pages-in-admin

#### Checklist

- [x] I've used the pull request template
- [x] I've linked this PR to the relevant issue (if mission work)
- [ ] I've written unit tests for these changes (if code change)
- [ ] I've updated the documentation in (If any documentation requires updating)
  - [ ] README.md
  - [ ] Elsewhere (please link)

### CODE REVIEW

File: app/forms/pages/conditions_form.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  attr_accessor :form, :page, :check_page_id, :routing_page_id, :answer_value, :goto_page_id
+
+  validates :answer_value, :goto_page_id, presence: true
+
+  def submit
+    return false if invalid?
+
+    Condition.create!(form_id: form.id,
+                      page_id: page.id,
+                      check_page_id: page.id, routing_page_id: page.id, answer_value:, goto_page_id:)
```
Comment (SamJamCul) [inline app/forms/pages/conditions_form.rb:14]: should this be separated out into one line per argument?

File: spec/factories/models/pages.rb [hunk 2]
```diff
@@ -10,7 +10,7 @@ def as_json(*args)
 
 FactoryBot.define do
   factory :page, class: "Page" do
-    association :form
+    id { Faker::Number.number(digits: 2) }
```
Comment (SamJamCul) [inline spec/factories/models/pages.rb:13]: does this risk overlapping with other pages generated within the same test?

**Review by SamJamCul — APPROVED** [review]
All good, just a little cosmetic thing really!

File: spec/factories/models/pages.rb (hunk 2 above)
Comment (DavidBiddle) [reply spec/factories/models/pages.rb:13]: Yes, but I'm not sure it matters at the moment - we aren't using it anywhere else, and in this PR it's only used to stop `goto_page_options` returning `id: nil` for every page in the form. But the tests will still pass if we remove this line so I don't mind removing it if we think the potential name clash is a concern.

Comment (aliuk2012) [reply spec/factories/models/pages.rb:13]: I think its fine for the moment, we will be visiting this again when we add more functionality

**Pull request merged**